pkg/gb/input.go

```diff
@@ -24,14 +26,61 @@ const (
 	ButtonDown = 7
 )
 
-// PressButton notifies the GameBoy that a button has just been pressed
+type ButtonInput struct {
+	// Pressed and Released are gameboy button inputs on this frame
+	Pressed, Released []Button
+
+	// KeysPressed list all keyboard keys pressed last frame
+	// This is used to call handlers that list/modify emulator config
+	KeysPressed []string
+}
+
+// IOBinding provides an interface for display and input bindings.
+type IOBinding interface {
+	// RenderScreen renders a frame of the game.
+	Render(screen *[160][144][3]uint8)
+	// ButtonInput returns which buttons were pressed and released
+	ButtonInput() ButtonInput
+	// SetTitle sets the title of the window.
+	SetTitle(title string)
+	// IsRunning returns if the monitor is still running.
+	IsRunning() bool
+}
+
+// pressButton notifies the GameBoy that a button has just been pressed
 // and requests a joypad interrupt.
-func (gb *Gameboy) PressButton(button Button) {
+func (gb *Gameboy) pressButton(button Button) {
 	gb.inputMask = bits.Reset(gb.inputMask, byte(button))
 	gb.requestInterrupt(4) // Request the joypad interrupt
 }
 
-// ReleaseButton notifies the GameBoy that a button has just been released.
-func (gb *Gameboy) ReleaseButton(button Button) {
+// releaseButton notifies the GameBoy that a button has just been released.
+func (gb *Gameboy) releaseButton(button Button) {
 	gb.inputMask = bits.Set(gb.inputMask, byte(button))
 }
+
+func (gb *Gameboy) ProcessInput(buttons ButtonInput) {
+
+	if !gb.IsGameLoaded() || gb.paused {
+		for _, pressedButton := range buttons.KeysPressed {
+			if pressedButton == "Escape" {
+				gb.keyHandlers[pressedButton]()
```

Comment:
I'm not sure we need this special case. It may be useful to pause emulation and then use some of the debugging inputs? You can probably just get rid of the whole `if !gb.IsGameLoaded() || gb.paused` block or wrap the `pressButton`/`releaseButton` calls in a check for if the gameboy is paused.